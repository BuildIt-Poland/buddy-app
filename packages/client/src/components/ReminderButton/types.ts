import { IconButtonProps } from '@material-ui/core/IconButton';

export interface ReminderButtonProps extends IconButtonProps {}
