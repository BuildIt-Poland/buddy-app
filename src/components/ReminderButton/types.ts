import { FabProps } from '@material-ui/core/Fab';

export interface ReminderButtonProps extends FabProps {}
