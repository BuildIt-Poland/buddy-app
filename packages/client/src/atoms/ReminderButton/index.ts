import ReminderButton from './ReminderButton';

export default ReminderButton;
