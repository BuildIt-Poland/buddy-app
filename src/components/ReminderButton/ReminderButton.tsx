import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import Fab from '@material-ui/core/Fab';
import NotificationsActiveOutlinedIcon from '@material-ui/icons/NotificationsActiveOutlined';
import { ReminderButtonProps } from './types';

const useStyles = makeStyles(theme => ({
  button: {
    margin: theme.spacing(1.5, 0),
  },
}));

const ReminderButton = (props: ReminderButtonProps) => {
  const { button } = useStyles();
  return (
    <Fab {...props} aria-label='reminder-btn' className={button}>
      <NotificationsActiveOutlinedIcon />
    </Fab>
  );
};

ReminderButton.defaultProps = {
  color: 'inherit',
  size: 'small',
};

export default ReminderButton;
