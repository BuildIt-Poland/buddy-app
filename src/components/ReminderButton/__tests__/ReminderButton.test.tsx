import React from 'react';
import { create } from 'react-test-renderer';

import ReminderButton from '../ReminderButton';

jest.mock('@material-ui/core/Fab', () => 'Fab');
jest.mock(
  '@material-ui/icons/NotificationsActiveOutlined',
  () => 'NotificationsActiveOutlinedIcon'
);

describe('Component - ReminderButton', () => {
  test('renders correctly', () => {
    const component = create(<ReminderButton />);

    expect(component.toJSON()).toMatchSnapshot();
  });
});
