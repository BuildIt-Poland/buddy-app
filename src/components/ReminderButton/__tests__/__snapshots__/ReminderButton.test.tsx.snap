// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component - ReminderButton renders correctly 1`] = `
<Fab
  aria-label="reminder-btn"
  className="makeStyles-button-1"
  color="inherit"
  size="small"
>
  <NotificationsActiveOutlinedIcon />
</Fab>
`;
